File: apps/desktop/src/types/group.ts
export interface ProfileGroup {
	id: string;
	name: string;
	sort_order: number;
	profile_count: number;
	created_at: string;
	updated_at: string;
}

export interface CreateGroupInput {
	name: string;
}

export interface UpdateGroupInput {
	name?: string;
	sort_order?: number;
}

File: apps/desktop/src-tauri/src/domain/group.rs
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone)]
pub struct ProfileGroup {
    pub id: String,
    pub name: String,
    pub sort_order: i32,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProfileGroupDto {
    pub id: String,
    pub name: String,
    pub sort_order: i32,
    pub profile_count: usize,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateGroupInput {
    pub name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateGroupInput {
    pub name: Option<String>,
    pub sort_order: Option<i32>,
}
